### PR TITLE
feat: update to use redis, fix #35

### LINKED ISSUE #35: Configuration files currently set to mongodb
Mongodb is deprecated -- config for all examples should be updated to use Redis.

### PR DESCRIPTION
**Pull request by charles-knox-intel**
Signed-off-by: charles-knox-intel <44684517+charles-knox-intel@users.noreply.github.com>

* Addresses issue #35, by updating all references of MongoDB to Redis
* Includes a couple basic readme formatting cleanup/grammar improvements

I did a limited validation sequence by bringing up the Geneva Redis compose stack, and running a few of the example services alongside them. The services start up and seem to work OK with the changes.

### CODE REVIEW

File: app-services/secrets/res/configuration.toml [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
         [Writable.InsecureSecrets.MongoDB.Secrets]
           username = ''
           password = ''
 
+      [Writable.InsecureSecrets.Redis]
+        Path = '/redis'
+        [Writable.InsecureSecrets.Redis.Secrets]
+          username = ''
+          password = ''
+
```
Comment (charles-knox-intel) [inline app-services/secrets/res/configuration.toml]: Need reviewers to confirm if MongoDB and Redis secrets can both simultaneously be defined in this way.

Comment (lenny-goodell) [reply app-services/secrets/res/configuration.toml]: Yes, they can. The path needs to be `redisdb`  w/o  `/`.  `/` also needs to be removed from `/mongodb`

File: app-services/secrets/res/configuration.toml [hunk 2]
```diff
@@ -1,30 +1,36 @@
...
-      
+
       [Writable.InsecureSecrets.MongoDB]
         Path = '/mongodb'
         [Writable.InsecureSecrets.MongoDB.Secrets]
           username = ''
           password = ''
 
+      [Writable.InsecureSecrets.Redis]
+        Path = '/redis'
```
Comment (lenny-goodell) [inline app-services/secrets/res/configuration.toml]: Change to `redisdb`

File: app-services/secrets/res/configuration.toml [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
+
       [Writable.InsecureSecrets.AWS]
         Path = '/aws'
         [Writable.InsecureSecrets.AWS.Secrets]
           username = 'aws-user'
           password = 'aws-pw'
-      
+
       [Writable.InsecureSecrets.MongoDB]
         Path = '/mongodb'
```
Comment (lenny-goodell) [inline app-services/secrets/res/configuration.toml]: change to `mongodb`

File: app-services/advanced-filter-convert-publish/res/configuration.toml [hunk 4]
```diff
@@ -26,9 +26,9 @@ EnableRemote = false
 File = './logs/simple-filter-xml-mqtt.log'
 
 [Database]
-Type = "mongodb"
+Type = "redis"
```
Comment (lenny-goodell) [inline app-services/advanced-filter-convert-publish/res/configuration.toml]: changed to `redisdb` here and in all other locations.

**Review by lenny-goodell — CHANGES_REQUESTED** [review]

**Review by lenny-goodell — APPROVED** [review]
LGTM

**Pull request merged**